src/basic/fs-util.h

```diff
@@ -132,6 +135,10 @@ int open_mkdir_at(int dirfd, const char *path, int flags, mode_t mode);
 
 int openat_report_new(int dirfd, const char *pathname, int flags, mode_t mode, bool *ret_newly_created);
 
+/* These are exposed to be overriden in src/shared/label.c */
+_weak_ int label_pre(int dir_fd, const char *path, mode_t mode);
+_weak_ int label_post(int dir_fd, const char *path);
```

Comment:
umpf. not a fan of `_weak_` stuff like this. this is ugly as fuck. `_weak_` might have its uses, but using it like that is wrong I am sure. this is just too magic, because what is linked into your program is just so hard to control.

If you want something like this, then make it explicit, i.e. define a structure `label_ops` or so, with two relevant function pointers in it (though I expect sooner or later there will be more). Do this in a new header file mac-util.h or so, and then in mac-util.c you add a global pointer variable to that struct, that is initially set to NULL.

Then, in mac-selinux.h you provide a "label_ops" instance with its own functions in it. And then in mac_selinux_init() you install that structure in the pointer.

in mac-util.h you#d then also provide static inline function wrappers that call the ops pointers if set, or return 0 otherwise.